sql/builtin_functions.cc

```diff
@@ -2513,6 +2517,127 @@ BoundFunction upper(const std::vector<BoundSqlExpression> & args)
+                    return ExpressionValue(bestScore,
+                                           args[0].getEffectiveTimestamp());
+
+
+                bool singleThreaded = query.size() + target.size() < 1000;
+
+                unsigned char convQuery[query.size()];
+                unsigned char convTarget[target.size()];
+
+                mutex myMutex;
+                atomic<int> idx(0);
+                map<char, int> charIdx;
+
+                auto fct = [&] (const string & in, unsigned char * out) {
+                    auto strLen = in.size();
+                    for (int i = 0; i < strLen; ++i) {
+                        auto & currChar = in[i];
+                        auto it = charIdx.find(currChar);
+                        if (it == charIdx.end()) {
+
+                            auto doEmplace = [&] ()
+                                {
+                                    auto res = charIdx.emplace(currChar, idx ++);
+                                    ExcAssert(std::get<1>(res));
+                                    it = std::get<0>(res);
+                                };
+
+                            if(singleThreaded) {
+                                doEmplace();
+                            }
+                            else {
+                                lock_guard<mutex> lock(myMutex);
+                                it = charIdx.find(currChar);
+                                if (it == charIdx.end()) {
+                                    doEmplace();
+                                }
+                            }
+                        }
+                        out[i] = it->second;
+                    }
+                };
+
+                if(singleThreaded) {
+                    fct(query, &convQuery[0]);
+                    fct(target, &convTarget[0]);
+                }
+                else {
+                    thread queryThread(fct, query, &convQuery[0]);
+                    thread targetThread(fct, target, &convTarget[0]);
+                    queryThread.join();
+                    targetThread.join();
+                }
+
+                // DEBUG OUTPUT------------------
+                /*
+                cerr << "AFTER JOIN" << endl;
+                cerr << query << endl;
+                for (int i = 0; i < query.size(); ++i) {
+                    cerr << (int)convQuery[i];
+                }
+                cerr << endl << endl << target << endl;
+                for (int i = 0; i < target.size(); ++i) {
+                    cerr << (int)convTarget[i];
+                }
+                cerr << endl;
+                for (const auto & it: charIdx) {
+                    cerr << it.first << " - " << it.second << endl;
+                }
+                */
+                // ------------------------------
+
+                int numLocations;
+                int* endLocations1, * startLocations1;
+                unsigned char* alignment; int alignmentLength;
```

Comment:
Lets put those two declarations on two different lines, they are not even of the same type.